codestyle/checkstyle.xml

```diff
@@ -204,6 +204,12 @@
       <property name="message" value="Use org.apache.druid.server.http.HttpMediaType#TEXT_PLAIN_UTF8 instead"/>
     </module>
 
+    <module name="Regexp">
+      <property name="format" value='^((?!throw|=).)* Throwables.propagate\('/>
+      <property name="illegalPattern" value="true"/>
+      <property name="message" value="Throw caught exception"/>
```

Comment:
I would say something like `Throwables.propagate() shouldn't be used in new code. See https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate`. We don't prohibit `Throwables.propagate()` right away via forbidden-apis only because there are so much uses in the codebase and that's a huge undertaking to fix them all.

Reply:
I can throw caught exception as is?

Reply:
If you are asking how to replace `Throwables.propagate()`, see [What to do](https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate#what-to-do) section in the article.

I suggest the following algorithm:
 1. Try to replace `Throwables.propagate(e)` with `throw e`. If it's fine and javac doesn't complain, then fine. It's also likely that IDE will then highlight possible simplification of code, like `catch (SomeException e) { throw e; }` is pointless and could be removed entirely.
 2. If `throw e` is not fine (IDE highlights something with red), replace it with `throw new RuntimeException(e)`.

Reply:
I'll follow the second solution as a best choice 👍

Reply:
I've updated my PR. However it fails at some tests for Exception type mismatch i.e.

`java.lang.AssertionError: 
Expected: (an instance of org.apache.druid.query.QueryInterruptedException and exception with cause an instance of java.util.concurrent.TimeoutException and exception with message a string containing "Cannot acquire enough merge buffers")
     but: an instance of org.apache.druid.query.QueryInterruptedException <java.lang.RuntimeException: java.lang.RuntimeException: QueryInterruptedException{msg=Cannot acquire enough merge buffers, code=Query timeout, class=java.util.concurrent.TimeoutException, host=null}> is a java.lang.RuntimeException
Stacktrace was: java.lang.RuntimeException: java.lang.RuntimeException: QueryInterruptedException{msg=Cannot acquire enough merge buffers, code=Query timeout, class=java.util.concurrent.TimeoutException, host=null}
	at org.apache.druid.java.util.common.guava.WrappingSequence.accumulate(WrappingSequence.java:61)
	at org.apache.druid.java.util.common.guava.MappedSequence.accumulate(MappedSequence.java:43)
	at org.apache.druid.java.util.common.guava.Sequence.toList(Sequence.java:76)
	at org.apache.druid.query.groupby.GroupByQueryRunnerTestHelper.runQuery(GroupByQueryRunnerTestHelper.java:58)
	at `

So, should I fix the tests which checks for `RuntimeException` or `throw e` for such cases @leventov ?

Reply:
Thanks for doing this work.

I suggest to replace that particular `throw new RuntimeException(e)` that leads to test breakage with
```java
Throwables.propagateIfPossible(e);
throw new RuntimeException(e);
```

Also please prohibit `Throwables.propagate()` in forbidden-apis.